src/sentry/relay/config/metric_extraction.py

```diff
@@ -271,9 +258,10 @@ def _merge_metric_specs(
 ) -> List[MetricSpec]:
     # We use a dict so that we can deduplicate metrics with the same hash.
     metrics: Dict[str, MetricSpec] = {}
-    for query_hash, spec in alert_specs + widget_specs:
+    for query_hash, spec, _ in alert_specs + widget_specs:
         already_present = metrics.get(query_hash)
         if already_present and already_present != spec:
+            # XXX: I do not think this should be an error
             logger.error(
```

Comment:
I think we should stop reporting this.

Reply:
We can't. If a query hash exists twice then the `count` / `distribution` table in snuba will have the wrong numbers, we've only seen it for Sentry alerts / dashboards but the small chance of collision exists for GA w/ more alerts and more widgets over time. 

When (not if) it happens that two specs come from both widget and alerts and have the same query_hash you have a couple possibilities:
1. Someone has a widget that matches alerts 1:1, same query hash, same specs. In this case, just de-duplicate and it's fine since you only want that count to be extracted once.
2.  Someone has a widget that matches the alert _query hash_ but **not** the same spec. This is what's happening when I added the env group-by tag. Technically in that case the group-by tag can supercede the alert spec, but if the specs are different, technically you always want separate query hashes.

#2 is really the problem, and part of the reason I wanted to clean up our query hashes for the environment group-by before GA.